crates/biome_css_analyze/src/lint/nursery/no_important_styles.rs

```diff
@@ -0,0 +1,128 @@
+use crate::CssRuleAction;
+use biome_analyze::context::RuleContext;
+use biome_analyze::{Ast, FixKind, Rule, RuleDiagnostic, RuleSource, declare_lint_rule};
+use biome_console::markup;
+use biome_css_factory::make::css_declaration;
+use biome_css_syntax::{CssDeclaration, CssDeclarationImportant};
+use biome_diagnostics::Severity;
+use biome_rowan::{AstNode, BatchMutationExt};
+
+declare_lint_rule! {
+    /// Disallow the use of the `!important` style.
+    ///
+    /// The `!important` CSS style is a declaration used to give a specific rule
+    /// higher precedence over other conflicting rules. When it is applied to a CSS
+    /// property, that property's value is prioritized over any other declarations,
+    /// regardless of specificity or order of appearance in the stylesheet.
+    ///
+    /// ### How `!important` Works
+    /// - Normally, CSS rules follow a cascade order, where the browser decides
+    ///   which rules apply based on specificity, inheritance, and proximity to the
+    ///   targeted element.
+    /// - Adding `!important` to a rule overrides this cascade logic, forcing the
+    ///   rule to apply even if other rules have higher specificity or are defined later.
+    ///
+    /// ### Why `!important` Should Be Avoided
+    /// While `!important` can solve specific and immediate styling issues, its effects
+    /// can result in long-term problems within a codebase:
+    ///
+    /// - **Breaks the Cascade Logic**: It overrides the natural flow of cascading rules,
+    ///   making it harder to predict which styles will apply.
+    /// - **Increases Complexity**: Once `!important` is used in a stylesheet, other developers
+    ///   may respond by using it even more aggressively, creating a cycle of overrides and
+    ///   increasing maintenance difficulty.
+    /// - **Reduces Reusability**: Overriding styles often makes components less flexible,
+    ///   as future adjustments require more effort.
+    /// - **Hinders Debugging**: Debugging styles becomes more challenging, as developers
+    ///   must account for the `!important` rule overriding expected behavior.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// .style {
+    ///     color: red !important;
+    /// }
+    /// ```
+    ///
+    /// ## Useful links
+    ///
+    /// - [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/important)
+    pub NoImportantStyles {
+        version: "next",
+        name: "noImportantStyles",
+        language: "css",
+        recommended: true,
+        severity: Severity::Warning,
+        sources: &[RuleSource::Stylelint("declaration-no-important")],
+        fix_kind: FixKind::Safe,
```

Comment:
I think this is an unsafe fix as the computed style can be unexpectedly changed.